test/functional/p2p_sendtxrcncl.py

```diff
@@ -152,39 +152,39 @@ def run_test(self):
         with self.nodes[0].assert_debug_log(['Forget txreconciliation state of peer']):
             peer.send_message(create_sendtxrcncl_msg())
             peer.send_message(msg_verack())
-        peer.peer_disconnect()
+        self.nodes[0].disconnect_p2ps()
 
         self.log.info('SENDTXRCNCL sent to an outbound')
         peer = self.nodes[0].add_outbound_p2p_connection(
-            SendTxrcnclReceiver(), wait_for_verack=True, p2p_idx=1, connection_type="outbound-full-relay")
+            SendTxrcnclReceiver(), wait_for_verack=True, p2p_idx=0, connection_type="outbound-full-relay")
         assert peer.sendtxrcncl_msg_received
         assert peer.sendtxrcncl_msg_received.initiator
         assert not peer.sendtxrcncl_msg_received.responder
         assert_equal(peer.sendtxrcncl_msg_received.version, 1)
-        peer.peer_disconnect()
+        self.nodes[0].disconnect_p2ps()
 
         self.log.info('SENDTXRCNCL should not be sent if block-relay-only')
         peer = self.nodes[0].add_outbound_p2p_connection(
-            SendTxrcnclReceiver(), wait_for_verack=True, p2p_idx=2, connection_type="block-relay-only")
+            SendTxrcnclReceiver(), wait_for_verack=True, p2p_idx=0, connection_type="block-relay-only")
         assert not peer.sendtxrcncl_msg_received
-        peer.peer_disconnect()
+        self.nodes[0].disconnect_p2ps()
 
         self.log.info("SENDTXRCNCL should not be sent if feeler")
-        peer = self.nodes[0].add_outbound_p2p_connection(P2PFeelerReceiver(), p2p_idx=2, connection_type="feeler")
+        peer = self.nodes[0].add_outbound_p2p_connection(P2PFeelerReceiver(), p2p_idx=0, connection_type="feeler")
         assert not peer.sendtxrcncl_msg_received
-        peer.peer_disconnect()
+        self.nodes[0].disconnect_p2ps()
 
         self.log.info('SENDTXRCNCL if block-relay-only triggers a disconnect')
         peer = self.nodes[0].add_outbound_p2p_connection(
-            PeerNoVerack(), wait_for_verack=False, p2p_idx=3, connection_type="block-relay-only")
+            PeerNoVerack(), wait_for_verack=False, p2p_idx=0, connection_type="block-relay-only")
         with self.nodes[0].assert_debug_log(["we indicated no tx relay; disconnecting"]):
             peer.send_message(create_sendtxrcncl_msg(initiator=False))
             peer.wait_for_disconnect()
```

Comment:
I think this is racy, so you'd have to call `self.nodes[0].disconnect_p2ps()` or use a unique index

Reply:
Can you explain why? We called `disconnect_p2ps()` before `add_outbound_p2p_connection`, so are guaranteed to have no peer when doing so. Within the`with` block we call `wait_for_disconnect` (which waits until the disconnect initiated by the node is completed), so the subsequent subtest  also shouldn't be able to run before the disconnect of the current one is completed. What am I missing?

Reply:
Oh, right. As the disconnect happens on the side of the node, there shouldn't be a race where the node is not aware of the disconnect.